packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts

```diff
@@ -0,0 +1,248 @@
+} from '../types';
+
+const IMPUTED_EXPIRY_SECONDS = 120;
+
+function getAltMarketInfo(
+    offerings: AltOffering[],
+    buyTokenAddress: string,
+    sellTokenAddress: string,
+): AltOffering | undefined {
+    for (const offering of offerings) {
+        if (
+            (buyTokenAddress === offering.baseAsset && sellTokenAddress === offering.quoteAsset) ||
+            (sellTokenAddress === offering.baseAsset && buyTokenAddress === offering.quoteAsset)
+        ) {
+            return offering;
+        }
+    }
+    return undefined;
+}
+
+function parseFirmQuoteResponseFromAltMM(altFirmQuoteReponse: AltFirmQuoteResponse): V4RFQFirmQuote {
+    return {
+        signedOrder: altFirmQuoteReponse.data['0xv4order'],
+    };
+}
+
+function parseIndicativeQuoteResponseFromAltMM(
+    altIndicativeQuoteReponse: AltIndicativeQuoteReponse,
+    altPair: AltOffering,
+    makerToken: string,
+    takerToken: string,
+): V4RFQIndicativeQuote {
+    let makerAmount: BigNumber;
+    let takerAmount: BigNumber;
+
+    if (!altIndicativeQuoteReponse.price) {
+        throw new Error('Price not returned by alt MM');
+    } else {
+        if (makerToken === altPair.baseAsset) {
+            if (altIndicativeQuoteReponse.amount) {
+                // base asset = maker asset, quote asset = taker asset, amount specified
+                makerAmount = Web3Wrapper.toBaseUnitAmount(
+                    new BigNumber(altIndicativeQuoteReponse.amount),
+                    altPair.baseAssetDecimals,
+                );
+                takerAmount = Web3Wrapper.toBaseUnitAmount(
+                    new BigNumber(altIndicativeQuoteReponse.amount)
+                        .times(new BigNumber(altIndicativeQuoteReponse.price))
+                        .decimalPlaces(altPair.quoteAssetDecimals, BigNumber.ROUND_DOWN),
+                    altPair.quoteAssetDecimals,
+                );
+            } else if (altIndicativeQuoteReponse.value) {
+                // base asset = maker asset, quote asset = taker asset, value specified
+                takerAmount = Web3Wrapper.toBaseUnitAmount(
+                    new BigNumber(altIndicativeQuoteReponse.value),
+                    altPair.quoteAssetDecimals,
+                );
+                makerAmount = Web3Wrapper.toBaseUnitAmount(
+                    new BigNumber(altIndicativeQuoteReponse.value)
+                        .dividedBy(new BigNumber(altIndicativeQuoteReponse.price))
+                        .decimalPlaces(altPair.baseAssetDecimals, BigNumber.ROUND_DOWN),
+                    altPair.baseAssetDecimals,
+                );
+            } else {
+                throw new Error('niether amount or value were specified');
+            }
+        } else if (makerToken === altPair.quoteAsset) {
+            if (altIndicativeQuoteReponse.amount) {
+                // base asset = taker asset, quote asset = maker asset, amount specified
+                takerAmount = Web3Wrapper.toBaseUnitAmount(
```

Comment:
Is there an opportunity to refactor these chunks of code that look fairly similar?